src/main/java/frc/robot/subsystems/vision/VisionSubsystem.java

```diff
@@ -1,175 +1,235 @@
-      Pose2d botPose = LimelightHelpers.getBotPose2d(currentlyUsedLimelight);
-      // The origin of botpose is at the center of the field
-      double robotX = botPose.getX() + FieldConstants.FIELD_LENGTH_METERS / 2.0;
-      double robotY = botPose.getY() + FieldConstants.FIELD_WIDTH_METERS / 2.0;
-      Rotation2d robotRotation = botPose.getRotation();
-      lastSeenPose = new Pose2d(robotX, robotY, robotRotation);
-      return new Pose2d(robotX, robotY, robotRotation);
+  public void updateLimelightPoseEstimate(int limelightNumber) {
+    if (!canSeeAprilTags(limelightNumber)) {
+      limelightEstimates[limelightNumber] = new PoseEstimate();
+    }
+
+    if (headingRateDegreesPerSecond < VisionConstants.MEGA_TAG_2_MAX_HEADING_RATE) {
+      LimelightHelpers.SetRobotOrientation(
+          getLimelightName(limelightNumber),
+          headingDegrees,
+          0, // <- dunno if this'll be useful
+          0,
+          0,
+          0,
+          0);
+      limelightEstimates[limelightNumber] =
+          LimelightHelpers.getBotPoseEstimate_wpiBlue_MegaTag2(getLimelightName(limelightNumber));
     } else {
-      return new Pose2d();
+      limelightEstimates[limelightNumber] =
+          LimelightHelpers.getBotPoseEstimate_wpiBlue(getLimelightName(limelightNumber));
     }
   }
 
   /**
-   * Returns the distance in meters from the limelight(s) to the closest 
-   * april tag that they can see.
+   * Gets the pose of the robot calculated by specified limelight via any April Tags it sees
+   *
+   * @param limelightNumber the number of the limelight
+   * @return the pose of the robot, if the limelight can't see any April Tags, it will return 0 for
+   *     x, y, and theta
    */
-  public double getDistanceFromClosestAprilTag() {
-    if (canSeeAprilTags()) {
-      int closestAprilTagID = (int) LimelightHelpers.getFiducialID(currentlyUsedLimelight);
-      double distance = getLimelightAprilTagDistance(closestAprilTagID);
-      // SmartDashboard.putNumber("distance from apriltag", distance);
-      return distance;
-    }
-    
-    // To be safe returns a big distance from the april tags if it can't see any
-    return Double.MAX_VALUE;
+  public Pose2d getPoseFromAprilTags(int limelightNumber) {
+    return limelightEstimates[limelightNumber].pose;
   }
 
   /**
-   * Returns how many april tags the limelight that is being used for pose
-   * estimation can see.
+   * Gets the Megatag1 pose of the robot calculated by specified limelight via any April Tags it
+   * sees
+   *
+   * @param limelightNumber the number of the limelight
+   * @return the Megatag1 pose of the robot, if the limelight can't see any April Tags, it will
+   *     return 0 for x, y, and theta
    */
-  public int getNumberOfAprilTags() {
-    return currentlyUsedLimelightResults.targetingResults.targets_Fiducials.length;
+  public Pose2d getMegatag1Pose(int limelightNumber) {
```

Comment:
I don't mind this, I think it helps clean up stuff, but if we have a method for mt1, shouldn't we also have a method for mt2?